icu4c/source/common/loclikely.cpp

```diff
@@ -826,7 +826,7 @@ createLikelySubtagsString(
     } \
 } UPRV_BLOCK_MACRO_END
 
-static void
+static UBool
```

Comment:
I find it confusing to introduce parallel error reporting paths, both the existing `UErrorCode` output parameter and the new `UBool` return value. Why not continue reporting all errors through `UErrorCode`?

Reply:
https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classicu_1_1Locale.html#a974d212572d7fc61a0579f820100a8f6 
said
"If this Locale is already in the maximal form, or not valid, or there is no data available for maximization, the Locale will be unchanged."
Notice, the case of "there is no data available for maximization, the Locale will be unchanged." - is not a condition of "error" from this API description.

The implementation of minimizeSubtags need to call addLikelySubtags, but the algorithm in 
http://www.unicode.org/reports/tr35/#Likely_Subtags stated
3. Return
3-1. If there is no match,either return
3-1-1. an error value, or ...

Because the above quoted text, addLikelySubtags won't return any error code for this case, but just return the Locale unchanged, based on the design of that API. That is not the problem, but ....

For minimizeSubtags, it stated:
1. First get max = AddLikelySubtags(inputLocale). If an error is signaled, return it.
to correctly implement the "If an error is signaled, return it." we need to catch this "error" - which is the "no match"
Therefore, I cannot use the UErrorCode to report the same information, because if I put any error code in the lower level, the upper level codes one for the public addLikelySubtags() and one for the internal usage inside minimizeSubtags won't be able to distinguish the error. For example, if I place U_ILLEGAL_ARGUMENT_ERROR into the UErrorCode, then the upper code cannot distinguish the U_ILLEGAL_ARGUMENT_ERROR caused by some other reason from the U_ILLEGAL_ARGUMENT_ERROR caused by "no matching".



minimizeSubtags
--

Reply:
Seems reasonable from what Frank says and what I see the call site doing.